packages/core/src/modules/x509/X509ModuleConfig.ts

```diff
@@ -1,22 +1,43 @@
+import type { AgentContext } from '../../agent'
+
+type GetTrustedCertificatesForProofOptions = {
+  proofRecordId?: string
+  correlationId?: string
+}
+
 export interface X509ModuleConfigOptions {
   /**
    *
    * Array of trusted base64-encoded certificate strings in the DER-format.
    */
   trustedCertificates?: [string, ...string[]]
+
+  getTrustedCertificatesForProof?(
```

Comment:
```suggestion
  /**
   * Optional callback method that will be called to dynamically get trusted certificates for a verification.
   * It will always provide the `agentContext` allowing to dynamically set the trusted certificates for a tenant.
   * If available the associated record id is also provided allowing to filter down trusted certificates to a single
   * exchange.
   */
  getTrustedCertificatesForVerification?(
```